docs/reference/commandline/build.md

```diff
@@ -489,6 +499,137 @@ FROM alpine AS production-env
+
+FROM scratch AS export-stage
+COPY --from=build-stage /go/bin/vndr /
+```
+
+When building the Dockerfile with the `-o` option, only the files from the final
+stage are exported to the `out` directory, in this case, the `vndr` binary:
+
+```bash
+$ docker build -o out .
+
+[+] Building 2.3s (7/7) FINISHED
+ => [internal] load build definition from Dockerfile                                                                          0.1s
+ => => transferring dockerfile: 176B                                                                                          0.0s
+ => [internal] load .dockerignore                                                                                             0.0s
+ => => transferring context: 2B                                                                                               0.0s
+ => [internal] load metadata for docker.io/library/golang:latest                                                              1.6s
+ => [build-stage 1/2] FROM docker.io/library/golang@sha256:2df96417dca0561bf1027742dcc5b446a18957cd28eba6aa79269f23f1846d3f   0.0s
+ => => resolve docker.io/library/golang@sha256:2df96417dca0561bf1027742dcc5b446a18957cd28eba6aa79269f23f1846d3f               0.0s
+ => CACHED [build-stage 2/2] RUN go get -u github.com/LK4D4/vndr                                                              0.0s
+ => [export-stage 1/1] COPY --from=build-stage /go/bin/vndr /                                                                 0.2s
+ => exporting to client                                                                                                       0.4s
+ => => copying files 10.30MB                                                                                                  0.3s
+
+$ ls ./out
+vndr
+```
+
+> **Note**: This feature requires the BuildKit backend. You can either
+> [enable BuildKit](../builder.md#buildkit) or use the [buildx](https://github.com/docker/buildx)
+> plugin which provides more output type options.
+
+### Specifying external cache sources
+
+In addition to local build cache, the builder can reuse the cache generated from
+previous builds with the `--cache-from` flag pointing to an image in the registry.
+
+To use an image as a cache source, cache metadata needs to be written into the
+image on creation. This can be done by setting `--build-arg BUILDKIT_INLINE_CACHE=1`
+when building the image. After that, the built image can be used as a cache source
+for subsequent builds.
+
+Upon importing the cache, the builder will only pull the JSON metadata from the
+registry and determine possible cache hits based on that information. If there
+is a cache hit, the matched layers are pulled into the local environment.
+
+In addition to images, the cache can also be pulled from special cache manifests
+generated by [`buildx`](https://github.com/docker/buildx) or the BuildKit CLI
+(`buildctl`). These manifests allow pulling layer data for intermediate stages
+in multi-stage builds.
```

Comment:
Can we clarify that these sentences are about `mode=max` cache?

Reply:
(Reading up on `mode=max`);

> - `mode=min (default): only export layers for the resulting image
> - `mode=max`: export all the layers of all intermediate steps. Not supported for inline cache exporter.

So, we should mention that when using `buildctl` (and `builds`?) this only works if those images were build by those using the `mode=max` option?

Reply:
Something like this?

```markdown
In addition to images, the cache can also be pulled from special cache manifests
 generated by [`buildx`](https://github.com/docker/buildx) or the BuildKit CLI
 (`buildctl`) when built with the `type=inline` and `mode=max` option.
```

(my only concern there would be that people would go looking for `mode=max` on the `docker` cli

Reply:
oh, probably only applies to that last sentence, so;

```markdown
These manifests (when built with the `type=inline` and `mode=max` options)
allow pulling layer data for intermediate stages in multi-stage builds.
```

Reply:
max is only supported for type=registry

Reply:
@AkihiroSuda @tonistiigi updated; PTAL